src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -1413,7 +1416,112 @@ private IBlockOperation CreateBoundBlockOperation(BoundBlock boundBlock)
             ITypeSymbol type = null;
             Optional<object> constantValue = default(Optional<object>);
             bool isImplicit = boundBlock.WasCompilerGenerated;
-            return new CSharpLazyBlockOperation(this, boundBlock, locals, _semanticModel, syntax, type, constantValue, isImplicit);
+            var parameters = (_semanticModel as MemberSemanticModel)?.MemberSymbol.GetParameters() ?? ImmutableArray<ParameterSymbol>.Empty;
+            var nullChecksToPrepend = GenerateNullChecksForParameters(parameters, syntax);
+            return new CSharpLazyBlockOperation(this, boundBlock, locals, nullChecksToPrepend, _semanticModel, syntax, type, constantValue, isImplicit);
+        }
+
+        private ConditionalOperation GenerateNullCheckForParameter(ParameterSymbol parameter, SyntaxNode syntax)
+        {
+            Optional<object> constantValue = default(Optional<object>);
+            var compilation = (CSharpCompilation)_semanticModel.Compilation;
+            IOperation conditionOp;
+            if (parameter.Type.IsNullableType())
+            {
+                conditionOp = new UnaryOperation(
+                    UnaryOperatorKind.Not,
+                    new InvocationOperation(
+                        targetMethod: ((IMethodSymbol)compilation.GetSpecialTypeMember(SpecialMember.System_Nullable_T_get_HasValue)).Construct(parameter.Type),
+                        instance: new ParameterReferenceOperation(parameter, _semanticModel, syntax, parameter.Type, constantValue, isImplicit: true),
+                        isVirtual: false,
+                        arguments: ImmutableArray<IArgumentOperation>.Empty,
+                        _semanticModel,
+                        syntax,
+                        compilation.GetSpecialType(SpecialType.System_Boolean),
+                        constantValue,
+                        isImplicit: true),
+                    isLifted: false,
+                    isChecked: false,
+                    operatorMethod: null,
+                    _semanticModel,
+                    syntax,
+                    compilation.GetSpecialType(SpecialType.System_Boolean),
+                    constantValue,
+                    isImplicit: true);
+            }
+            else
+            {
+                conditionOp = new BinaryOperation(BinaryOperatorKind.Equals,
```

Comment:
>BinaryOperatorKind.Equals, [](start = 50, length = 26)

Minor nit: please align all arguments at the same indentation like the lists above.